MediaGalleryUi/view/adminhtml/web/js/image/image-edit.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery',
+    'underscore',
+    'uiComponent',
+    'Magento_MediaGalleryUi/js/action/getDetails'
+], function ($, _, Component, getDetails) {
+    'use strict';
+
+    return Component.extend({
+        defaults: {
+            template: 'Magento_MediaGalleryUi/image/image-edit',
+            modalSelector: '',
+            imageEditDetailsUrl: '/media_gallery/image/details',
+            saveDetailsUrl: '/media_gallery/image/saveDetails',
+            images: [],
+            tagListLimit: 7,
+            showAllTags: false,
+            image: null,
+            modules: {
+                mediaGridMessages: '${ $.mediaGridMessages }'
+            }
+        },
+
+        /**
+         * Init observable variables
+         *
+         * @return {Object}
+         */
+        initObservable: function () {
+            this._super()
+                .observe([
+                    'image',
+                    'showAllTags'
+                ]);
+
+            return this;
+        },
+
+        /**
+         * Get image details by ID
+         *
+         * @param {String} imageId
+         */
+        getImageDetailsById: function (imageId) {
```

Comment:
Let's rename the method to reflect the functionality better
```suggestion
        showEditDetailsPanel: function (imageId) {
```